projects/plugins/jetpack/extensions/blocks/calendly/test/edit.js

```diff
@@ -0,0 +1,182 @@
+			} );
+		} )
+	} )
+);
+describe( 'CalendlyEdit', () => {
+	const defaultAttributes = {
+		backgroundColor: '#ffffff',
+		hideEventTypeDetails: false,
+		primaryColor: '#7d7d7d',
+		textColor: '#000000',
+		style: 'inline',
+		url: 'https://calendly.com/username',
+	};
+
+	const createErrorNotice = jest.fn();
+	const removeAllNotices = jest.fn();
+	const setAttributes = jest.fn();
+
+	const defaultProps = {
+		attributes: defaultAttributes,
+		setAttributes,
+		className: '',
+		clientId: 1,
+		name: 'jetpack/calendly',
+		noticeOperations: {
+			removeAllNotices,
+			createErrorNotice,
+		},
+	};
+
+	const propsWithoutUrl = {
+		...defaultProps,
+		attributes: {
+			...defaultAttributes,
+			url: '',
+		},
+	};
+
+	beforeEach( () => {
+		createErrorNotice.mockClear();
+		removeAllNotices.mockClear();
+		setAttributes.mockClear();
+	} );
+
+	test( 'validates block attributes', () => {
+		const attributes = { ...defaultAttributes, invalid: true };
+
+		render( <CalendlyEdit { ...{ ...defaultProps, attributes } } /> );
+
+		expect( setAttributes ).toHaveBeenCalledWith( defaultAttributes );
+	} );
+
+	test( 'set undefined url and displays error when invalid url supplied', async () => {
+		const attributes = { ...defaultAttributes, url: 'https://calendly.com/invalid-url' };
+		render( <CalendlyEdit { ...{ ...defaultProps, attributes } } /> );
+
+		expect( testEmbedUrl ).toHaveBeenCalledWith( attributes.url, expect.anything() );
+
+		await waitFor( () => expect( setAttributes ).toHaveBeenCalledWith( { url: undefined } ) );
+		expect( removeAllNotices ).toHaveBeenCalled();
+		expect( createErrorNotice ).toHaveBeenCalled();
+	} );
+
+	describe( 'parseEmbedCode', () => {
+		test( 'displays error notice when empty embed url submitted', () => {
+			render( <CalendlyEdit { ...propsWithoutUrl } /> );
+
+			userEvent.click( screen.getByRole( 'button', { name: 'Embed' } ) );
+
+			expect( removeAllNotices ).toHaveBeenCalled();
+			expect( createErrorNotice ).toHaveBeenCalled();
+		} );
+
+		test( 'displays error notice when updated embed code fails to parse', () => {
+			render( <CalendlyEdit { ...propsWithoutUrl } /> );
+
+			userEvent.paste(
+				screen.getByPlaceholderText( 'Calendly web address or embed code…' ),
+				'invalid-url'
+			);
+			userEvent.click( screen.getByRole( 'button', { name: 'Embed' } ) );
+
+			expect( removeAllNotices ).toHaveBeenCalled();
+			expect( createErrorNotice ).toHaveBeenCalled();
+		} );
+
+		test.skip( 'updates inner button block when new link embed code parsed', () => {
+			// TODO: Work out how to test this dispatched action.
+		} );
+
+		test( 'parsed embed code is tested before updating attributes', async () => {
+			render( <CalendlyEdit { ...propsWithoutUrl } /> );
+
+			userEvent.type( screen.getByRole( 'textbox' ), 'https://calendly.com/valid-url' );
+			userEvent.click( screen.getByRole( 'button', { name: 'Embed' } ) );
+			await waitFor( () =>
```

Comment:
because the state update in this case is dependent on a promise resolution wrapping it in an waitFor gets rid if the need for an `act` wrapper as in these cases the react testing library automatically handles the act wrapping - https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning